src/Orchard.Web/Modules/Orchard.CustomForms/Controllers/ItemController.cs

```diff
@@ -198,6 +198,13 @@ public class ItemController : Controller, IUpdateModel {
 
             contentItem.As<ICommonPart>().Container = customForm.ContentItem;
 
+            // save the submitted form
+            if (customForm.SaveContentItem) {
+                conditionallyPublish(contentItem);
+                // Refresh content item
+                contentItem = _contentManager.Get(contentItem.Id);
```

Comment:
Does this have any effect? It should just return the same instance from the cache. Or is the publishing creating a new version.

Reply:
The publish procedure is triggering the drivers of all the fields and creating a new version so I noticed I get the "compiled" content only by reading it again from the content manager, because there there may be some LazyField inside the content item.

Before editing this controller, I experimented inside the razor executed in the workflow and the only way to make it work was to force the content to be published and reading it again from the content manager.

Reply:
LGTM based on your answers! One question for line 205: If the item is conditionally published (line 203), then `_contentManager.Get(contentItem.Id)` will return null, if the item is not actually published, so versioning needs to be taken into account.

Reply:
It makes sense, thanks; suggestion tested on my environment and applied on current pr branch.